components/ListItem.js

```diff
@@ -1,25 +1,36 @@
+import Link from "next/link";
 import { styled } from "styled-components";
 
-export default function ListItem({ id, name, breed, birthday }) {
-  const date = new Date();
-  const birthdayDate = new Date(birthday);
-  const difference = date.getTime() - birthdayDate.getTime();
-  let months = Math.floor(difference / 2629746000);
-  const years = Math.floor(months / 12);
-  months = months - 12 * years;
-  const age = `${years}.${months} years`;
+export default function ListItem({
+  id,
+  name,
+  breed,
+  birthday,
+  slug,
+  calculateAge,
+}) {
+  const age = calculateAge(birthday);
```

Comment:
does this function needs to be a prop or can you put it in a utils folder and simply import the function in this file?

Reply:
thx